### DISCUSSION TITLE
Update XRechnung V2.3.1

### DISCUSSION THREAD

**Discussion by aschliewert** [stephanstapel/ZUGFeRD-csharp#214] [General]
Hello. Will there be an update for version 2.3.1 of XRechnung?

**Comment by stephanstapel** [comment]
Depending on the demand of the community. What are you missing?
Doesn't sound like a big thing:

![grafik](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/2912080/4d6c67a0-a005-4b93-aa5c-bb91f230f439)

**Reply by aschliewert** [reply]
Hi. Sorry for the late reply, I was on vacation. I am missing the C.1. changes.

**Comment by stephanstapel** [comment]
So just the id identifier in GuidelineSpecifiedDocumentContextParameter?

**Reply by stephanstapel** [reply]
like this:

![grafik](https://github.com/stephanstapel/ZUGFeRD-csharp/assets/2912080/c10d9f0f-e16e-4b87-b102-707c1f1af625)

?

**Reply by aschliewert** [reply]
Yes like this.

**Reply by aschliewert** [reply]
When is this update coming?

**Reply by stephanstapel** [reply]
Thanks for the reminder. It is pushed now into the repo, nuget package is out

**Reply by aschliewert** [reply]
Thank you very much!

**Comment by aschliewert** [comment]
Profile.cs


case ZUGFeRDVersion.Version21:
                    switch (profile)
                    {
                        case Profile.Minimum: return "urn:factur-x.eu:1p0:minimum";
                        case Profile.Basic: return "urn:cen.eu:en16931:2017#compliant#urn:factur-x.eu:1p0:basic";
                        case Profile.BasicWL: return "urn:factur-x.eu:1p0:basicwl";
                        case Profile.Comfort: return "urn:cen.eu:en16931:2017";
                        case Profile.Extended: return "urn:cen.eu:en16931:2017#conformant#urn:factur-x.eu:1p0:extended";
                        case Profile.XRechnung1: return "urn:cen.eu:en16931:2017#compliant#urn:xoev-de:kosit:standard:xrechnung_1.2";
                        case Profile.XRechnung:
                            {
                                **if (DateTime.Now >= new DateTime(2022, 08, 01))
                                {
                                    return "urn:cen.eu:en16931:2017#compliant#urn:xoev-de:kosit:standard:xrechnung_2.2";
                                }
                                else
                                {
                                    return "urn:cen.eu:en16931:2017#compliant#urn:xoev-de:kosit:standard:xrechnung_2.1";
                                }**
                            }
                        default: throw new Exception("Unsupported profile for ZUGFeRD version 21");
                    }                    
                    
                  
   This needs to be changed too, or am I wrong?

**Comment by stephanstapel** [comment]
That is exactly the place where the was changed:

https://github.com/stephanstapel/ZUGFeRD-csharp/blob/master/ZUGFeRD/Profile.cs

Or did I misunderstand your question?

**Reply by aschliewert** [reply]
Oh my mistake. I accidentally still had an older version of the repo in the same directory on my PC. ´:-D
